airflow/models/baseoperator.py

```diff
@@ -1001,6 +1001,14 @@ def __enter__(self):
     def __exit__(self, exc_type, exc_val, exc_tb):
         SetupTeardownContext.set_work_task_roots_and_leaves()
 
+    @staticmethod
+    def add_ctx_task(task):
```

Comment:
I'm a little torn on this name. Maybe `add_task_to_ctx`?

Are we planning on doing the same thing for adding tasks to a TaskGroup?

Can we check that this isn't being used outside a context manager?

Reply:
I will do the same for TaskGroup after this PR